File: reverse_date/reverse_date.rb
require 'date'

module ReverseDate
  def self.start(from, to)
    reverse_dates = []
    target_dates = make_and_process_date(from, to)
    target_dates.each do |d|
      next if d.to_s[-1].to_i % 2 == 0
      reverse_dates << d if reverse_date?(d)
    end
    p reverse_dates
  end

  def self.make_and_process_date(from, to)
    date = Date.parse(from)..Date.parse(to)
    date = date.map{ |d| d.strftime('%Y%m%d').to_i}
  end

  def self.reverse_date?(target)
    target == target.to_s(2).reverse.to_i(2)
  end
end

ReverseDate.start('19641010', '20200724')

File: reverse_date/reverse_date.js
String.prototype.reverse = function() {
    return this.split('').reverse().join('');
};

var ReverseDate = {
    start: function() {
        var reverseDates = [];

        while (true) {
            var date = this.dateFormat(this.targetDate, 'YYYYmmdd');
            if (this.isReverseDate(date)) {
                reverseDates.push(date);
            }
            if (date == 20200724) {
                break;
            }
            this.nextDate();
        }
        console.log(reverseDates);

    },
    targetDate: new Date('1964/10/10'),
    dateFormat: function (date, format) {
        var result = format;
        for (var key in this.fmt)
            result = result.replace(key, this.fmt[key](date));
        return parseInt(result);
    },
    fmt : {
        dd: function (date) {
            return ('0' + date.getDate()).slice(-2);
        },
        YYYY: function (date) {
            return date.getFullYear() + '';
        },
        mm: function (date) {
            return ('0' + (date.getMonth() + 1)).slice(-2);
        }
    },
    nextDate: function () {
        this.targetDate.setDate(this.targetDate.getDate() + 1);
    },
    isReverseDate: function(date) {
        return date == parseInt(date.toString(2).reverse(), 2);
    }
};

ReverseDate.start();
